salt/config/__init__.py

```diff
@@ -1065,6 +1072,10 @@
     'event_match_type': 'startswith',
     'minion_restart_command': [],
     'pub_ret': True,
+    'proxy_host': None,
```

Comment:
Personally, to me foo = '' is set, just empty
foo = None exists but has no value

with proxy_host setting it to '' is never going to work

but then consider it with proxy_password = '' could be a valid password.
It would then allow you to differentiate between an empty password and i
dont want to supply a password in this example.

To be honest is mostly personal preference, id rather be explicit (perhaps
it stems from a perl background) where you would do:
my $foo;
vs
my $foo = '';

The first initializes the variable for use and the latter assigns a value
to it.

But hey the approach you’ve mentioned is consistent with the other code.


Cheers
Brendan


On Wed, Jul 6, 2016 at 2:40 PM, Mike Place <notifications@github.com> wrote:

> In salt/config/__init__.py
> <https://github.com/saltstack/salt/pull/34407#discussion_r69674615>:
>
> > @@ -1065,6 +1072,10 @@
> >      'event_match_type': 'startswith',
> >      'minion_restart_command': [],
> >      'pub_ret': True,
> > +    'proxy_host': None,
>
> Out of curiosity, why not? I'd certainly be interested in a good reason. :]
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/saltstack/salt/pull/34407/files/307a3d94ad7e1b0d4ea812f0ffdb5a5b0783dbe2#r69674615>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAOjDQsnizvgMvwxKCmKDdSD4sRUP8oXks5qSzGngaJpZM4JCsTj>
> .
>